key/group.go

```diff
@@ -454,7 +461,11 @@ func (g *Group) ToProto(version commonutils.Version) *proto.GroupPacket {
 	out.SchemeID = g.Scheme.ID
 
 	out.Metadata = common.NewMetadata(version.ToProto())
-	out.Metadata.BeaconID = g.ID
+	if g.ID == "" {
+		out.Metadata.BeaconID = commonutils.DefaultBeaconID
```

Comment:
I see this method so many times - please let's refactor and put this into its own function, stg like `getOrDefault(id)`